src/Razor/src/Microsoft.VisualStudio.LiveShare.Razor/Guest/DefaultProxyAccessor.cs

```diff
@@ -56,7 +53,7 @@ public override IProjectHierarchyProxy GetProjectHierarchyProxy()
         internal virtual TProxy CreateServiceProxy<TProxy>() where TProxy : class
         {
 #pragma warning disable VSTHRD110 // Observe result of async calls
-            return _joinableTaskFactory.Run(() => _liveShareSessionAccessor.Session?.GetRemoteServiceAsync<TProxy>(typeof(TProxy).Name, CancellationToken.None));
+            return _joinableTaskFactory.Run(() => _liveShareSessionAccessor.Session?.GetRemoteServiceAsync<TProxy>(typeof(TProxy).Name, CancellationToken.None)!);
```

Comment:
This is a bit scary. If the code was allowing for Session to be null, why the `!`? Isn't the return type wrong?

Perhaps this should at least have a Debug.Assert so we might find issues, or an Assumes.NotNull if we're confident?

In general I'm wary of putting null suppression in when enabling nullable, but particularly for return values it's not ideal. If this assumption is wrong then tracking it down to here will be very very hard.

Reply:
I've looked through code paths and it seems for me that the null check of the `Session` is misleading on its own. First, the resut of this method is later used "as is" without any additional null checks, so even if we check `Session` here, we just bury the NRE a bit deeper in the callstack. Second, just logically why would you create a guest proxy if your session does not exists (is null in other words)? If such situation happens, the call of this method is the issue. So I think, assuming that the result after such operations is not null is kinda fighting with the consequence, instead of with the real cause of the problem. I would say, assuming the `Session` itself is not null is a more preferable check here and in the case below and I'm gonna do it. If you don't agree, feel free to reopen this review, I am open for a discussion